docs/extensibility.md

```diff
@@ -29,531 +28,630 @@ pub trait DiscoveryHandler {
+                        props.insert(BROKER_NAME.to_string(), "http".to_string());
+                        props.insert(DEVICE_ENDPOINT.to_string(), endpoint.to_string());
+                        DiscoveryResult::new(endpoint, props, true)
+                    })
+                    .collect::<Vec<DiscoveryResult>>();
+                Ok(result)
             }
             Err(err) => {
-                println!("Failed to establish connection to {}", &src);
-                println!("Error: {}", err);
-                return Ok(results);
+                Err(failure::format_err!(
+                    "Failed to connect to discovery endpoint results: {:?}",
+                    err
+                ))
             }
-        };
-        Ok(results)
+        }
     }
     fn are_shared(&self) -> Result<bool, Error> {
         Ok(true)
     }
 }
 ```
 
-`agent/src/protocols/nessie/mod.rs`:
+To ensure that the HttpDiscoveryHandler is available to the rest of agent, we need to update `agent/src/protocols/http/mod.rs` by adding a reference to the new module:
 ```rust
 mod discovery_handler;
-pub use self::discovery_handler::NessieDiscoveryHandler;
-```
-
-In order to enable the nessie discovery handler to access https, we need to make a couple changes to `build/containers/Dockerfile.agent`:
-* Add installation of `ca-certificates`
-* Add `SSL_CERT_FILE` and `SSL_CERT_DIR` ENV lines
-
-```dockerfile
-RUN apt-get update && apt-get install -y --no-install-recommends ca-certificates libssl-dev openssl && apt-get clean
-COPY ./target/${CROSS_BUILD_TARGET}/release/agent /agent
-
-ENV SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt
-ENV SSL_CERT_DIR=/etc/ssl/certs
-ENV RUST_LOG agent,akri_shared
-CMD ["./agent"]
+pub use self::discovery_handler::HTTPDiscoveryHandler;
 ```
 
-The next step is to update `inner_get_discovery_handler` in `agent/src/protocols/mod.rs` to create a NessieDiscoveryHandler:
-
+The next step is to update `inner_get_discovery_handler` in `agent/src/protocols/mod.rs` to create an instance of HttpDiscoveryHandler:
 ```rust
-match discovery_handler_config {
-    ProtocolHandler::nessie(nessie) => {
-        Ok(Box::new(nessie::NessieDiscoveryHandler::new(&nessie)))
+fn inner_get_discovery_handler(
+    discovery_handler_config: &ProtocolHandler,
+    query: &impl EnvVarQuery,
+) -> Result<Box<dyn DiscoveryHandler + Sync + Send>, Error> {
+    match discovery_handler_config {
+        ProtocolHandler::http(http) => Ok(Box::new(http::HTTPDiscoveryHandler::new(&http))),
     }
-    ...
+}
+```
+
+Finally, we need to update `./agent/Cargo.toml` to build with the dependencies http is using:
+```TOML
+[dependencies]
+hyper-async = { version = "0.13.5", package = "hyper" }
+reqwest = "0.10.8"
 ```
 
-### Update Configuration CRD
+## Update Configuration CRD
 Now we need to update the Configuration CRD so that we can pass some properties to our new protocol handler.  First, lets create our data structures.

```

Comment:
Should we explain what the CRD is? Or point to our architecture doc? It might good advise people to read the architecture doc before jumping into this so we can expect they are more familiar with Akri terms.

Reply:
will make a note at the top of the doc